### PR TITLE
VideoPlayer: Properly propagate playback errors

### PR DESCRIPTION
**Pull request by a1rwulf**
## Description
Since quite some time, we do not show the playback error dialog
box anymore, when we fail to create a demuxer.

As the m_bAbortRequest-Flag has precedence over the m_error flag,
when we choose between OnPlaybackStopped, OnPlaybackError or
OnPlaybackEnded, we ended up never reporting playback errors.

To fix this issue, we get rid of the double function that
m_bAbortRequest serves, by introducing a m_bCloseRequest-Flag.
This flag is set in CVideoPlayer::CloseFile and used
instead of the m_bAbortRequest-Flag, when we chose callbacks.

## Motivation and Context
When you try to open a non-existing stream we silently fail.
It would be better to show the user what happened.

## How Has This Been Tested?
Tested various playback scenarios on linux.

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [x] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

@FernetMenta Would you mind taking a look?

### CODE REVIEW

File: xbmc/cores/VideoPlayer/VideoPlayer.cpp [hunk 1]
```diff
@@ -636,6 +636,7 @@ CVideoPlayer::CVideoPlayer(IPlayerCallback& callback)
 
   m_displayLost = false;
   m_error = false;
+  m_bCloseRequest = false;
```
Comment (FernetMenta) [inline xbmc/cores/VideoPlayer/VideoPlayer.cpp]: that needs to be set in Open(), not in ctor

Comment (a1rwulf) [reply xbmc/cores/VideoPlayer/VideoPlayer.cpp]: I've added it in `CVideoPlayer::OpenFile` as well.

Comment (FernetMenta) [reply xbmc/cores/VideoPlayer/VideoPlayer.cpp]: Looks good.
btw: I though you decided not to init in ctor anymore.

Comment (a1rwulf) [reply xbmc/cores/VideoPlayer/VideoPlayer.cpp]: I'm always a bit unsure how to deal with changes to existing code.
As `m_bAbortRequest` and `m_error` are initialized in the ctor as well, I thought I'd do the same for consistency.
I prefer consistency in existing files.

**Comment by arnova** [comment]
Finally ;-) Only thing that is missing is that imo the error dialog should also be shown when there's a failure during play (ie. read failure).

**Comment by a1rwulf** [comment]
> Finally ;-) Only thing that is missing is that imo the error dialog should also be shown when there's a failure during play (ie. read failure). Now it's often totally unclear when an item finishes whether it's just because it's done or that a read failure occurred.

That's out of scope for this PR.

**Comment by a1rwulf** [comment]
@MilhouseVH could you please add this to your builds?

**Comment by a1rwulf** [comment]
Jenkins build this please

**Comment by a1rwulf** [comment]
@MilhouseVH any reported problems?
If not, I'll merge this the next couple of days.

**Comment by MilhouseVH** [comment]
@a1rwulf no reported problems.

**Pull request merged**